packages/accounts-controller/src/AccountsController.ts

```diff
@@ -241,12 +264,53 @@ export class AccountsController extends BaseController<
   }
 
   /**
-   * Returns the selected internal account.
+   * Returns the selected evm internal account by default unless the namespace is not eip155
    *
+   * @param chainId - Caip2 Id of the account
    * @returns The selected internal account.
    */
-  getSelectedAccount(): InternalAccount {
-    return this.getAccountExpect(this.state.internalAccounts.selectedAccount);
+  getSelectedAccount(chainId: Caip2ChainId = 'eip155:*'): InternalAccount {
+    if (!isCaipChainId(chainId) && chainId !== 'eip155:*') {
+      throw new Error(`Invalid CAIP2 id ${String(chainId)}`);
+    }
+
+    // TODO: have CAIP2 addresses within InternalAccount
+    const selectedAccount = this.getAccountExpect(
+      this.state.internalAccounts.selectedAccount,
+    );
+
+    if (
+      !chainId.startsWith('eip155:') ||
+      (chainId.startsWith('eip155:') &&
+        selectedAccount.type.startsWith('eip155:'))
+    ) {
+      return selectedAccount;
+    }
+
+    const accounts = this.listAccounts().filter((account) =>
+      account.type.startsWith('eip155:'),
+    );
+    let lastSelectedEvmAccount = accounts[0];
```

Comment:
We should check for `.length` here before using `[0]`. So we could move this block here:

```ts
// if (!lastSelectedEvmAccount) {
if (!accounts.length) {
  // !Should never reach this.
  throw new Error('AccountsController: No evm accounts');
}
let lastSelectedEvmAccount = accounts[0];
```